beacon-chain/blockchain/service.go

```diff
@@ -300,11 +302,17 @@ func (c *ChainService) blockProcessing(done <-chan struct{}) {
 				log.Debugf("Incoming block failed validity conditions: %v", err)
 			}
 
-			// Process attestation as a validator.
-			if err := c.chain.processAttestations(block); err != nil {
-				// We might receive a lot of blocks that fail attestation processing,
-				// so we create a debug level log instead of an error log.
-				log.Debugf("could not process attestation: %v", err)
+			// Process attestations as a beacon chain node.
+			for index := range block.Attestations() {
+				if err := c.chain.processAttestation(index, block); err != nil {
+					// We might receive a lot of blocks that fail attestation processing,
+					// so we create a debug level log instead of an error log.
+					log.Debugf("could not process attestation: %v", err)
```

Comment:
Why doesn't this immediately fail and exit? Isn't a block with invalid attestations an invalid block?

Reply:
I don't think that's true. A block with can have many attestations. Each attestation is a bundle of votes. If a Vote from shard 45 for Block Hash ABCD is invalid, that doesn't mean we should skip processing rest of the attestations

Reply:
Excluding invalid attestations from the active state is less trivial. I still feel like we should invalidate the entire block if a single attestation is invalid. In what situation would an honest note submit a block with an invalid attestation? If an attester sends an invalid signature to a proposer, it should be the proposer's responsibility to ignore it.